### PR TITLE
Improve insertText handling of inline elements

### PR DESCRIPTION
**Pull request by trueadm**
This improves on the heuristics of `insertText`, so that it better handles inline elements. Working with @acywatson on this, so it might need some further work.

### CODE REVIEW

File: packages/lexical/src/LexicalSelection.js [hunk 1]
```diff
@@ -848,55 +863,60 @@ export class RangeSelection implements BaseSelection {
...
+
+        if (lastNodeChild.isAttached()) {
           if (
-            lastNodeChild.is(firstNode) ||
-            ($isElementNode(lastNodeChild) &&
-              lastNodeChild.isParentOf(firstNode))
+            !selectedNodesSet.has(lastNodeChild) ||
+            lastNodeChild.is(lastNode) ||
+            (lastElementWasInline &&
+              lastNodeChild.is(lastNode.getParentOrThrow()))
```
Comment (acywatson) [inline packages/lexical/src/LexicalSelection.js:891]: The use of lastNode seems a little confusing here. For instance,``` lastNodeChild.is(lastNode.getParentOrThrow())``` kind of suggests that a child of "lastNode" and be the same as the parent of "lastNode".  I think it goes back to line 862 where we start to call the children of lastElement "lastNodeChildren".  Maybe they should be lastElementChildren?

File: packages/lexical/src/LexicalSelection.js [hunk 2]
```diff
@@ -800,12 +800,27 @@ export class RangeSelection implements BaseSelection {
...
+      // We have to get the parent elements before the next section,
+      // as in that section we might mutate the lastNode.
       const firstElement = $isElementNode(firstNode)
         ? firstNode
         : firstNode.getParentOrThrow();
-      const lastElement = $isElementNode(lastNode)
+      let lastElement = $isElementNode(lastNode)
         ? lastNode
         : lastNode.getParentOrThrow();
+      const shouldInsertAfterFirstElement =
```
Comment (acywatson) [inline packages/lexical/src/LexicalSelection.js:811]: I think we only use this constant once. If we want to keep if for readability, that's probably OK. But given how it's used below it seems to me we might as well inline the expression.

**Review by acywatson — APPROVED** [review]
Seems to solve the immediate problem(s), couple of thoughts around the naming.

File: packages/lexical/src/LexicalSelection.js (hunk 1 above)
Comment (trueadm) [reply packages/lexical/src/LexicalSelection.js:895]: Addressed with a comment. :)

File: packages/lexical/src/LexicalSelection.js (hunk 2 above)
Comment (trueadm) [reply packages/lexical/src/LexicalSelection.js]: Agreed, moved it inline.

**Pull request merged**